lab 01 updated, first commit

Marchitect opened the pull request:
Fixes #439 

Review of colleague's PR #431 

#### Changes proposed in this PR:

 - updated abs() and fib() function to generate Fibonacci and Negafibonacci numbers
 - updated test cases

Marchitect committed:
update lab 01 first commit

juliaogris reviewed (changes requested):
Nice work!! This must have been one of the best initial PRs I've seen so far.

Please [fix commit messages](https://github.com/anz-bank/go-course/wiki/Lab-1:-Collected-Code-Review-Comments#commit-messages) with `git rebase -i master`.

FWIW, your solution is very close to what we came up with as the canonical solution after a bit of back and forth. In case you are interest here's ours:
```go
func fib(n int) {
	sign := 1
	if n < 0 {
		sign = -1
	}

	n1, n2 := 0, 1
	for j := 0; j < n*sign; j++ {
		n1, n2 = n2, n1+sign*n2
		fmt.Fprintln(out, n1)
	}
}
```

You could use a map rather than a slice for your table tests with the key being the test case name which can (independently of data structure) also be used as the name of the sub test.
See [Dave Cheney's post](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests) for more details if you are curious (no need to fix anything, but don't hold back if you feel the urge ;) ).

Marchitect committed:
update lab 01 testing file, replace table with map based testcases

Marchitect commented:
> Nice work!! This must have been one of the best initial PRs I've seen so far.
> 
> Please [fix commit messages](https://github.com/anz-bank/go-course/wiki/Lab-1:-Collected-Code-Review-Comments#commit-messages) with `git rebase -i master`.
> 
> FWIW, your solution is very close to what we came up with as the canonical solution after a bit of back and forth. In case you are interest here's ours:
> 
> ```go
> func fib(n int) {
> 	sign := 1
> 	if n < 0 {
> 		sign = -1
> 	}
> 
> 	n1, n2 := 0, 1
> 	for j := 0; j < n*sign; j++ {
> 		n1, n2 = n2, n1+sign*n2
> 		fmt.Fprintln(out, n1)
> 	}
> }
> ```
> 
> You could use a map rather than a slice for your table tests with the key being the test case name which can (independently of data structure) also be used as the name of the sub test.
> See [Dave Cheney's post](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests) for more details if you are curious (no need to fix anything, but don't hold back if you feel the urge ;) ).

Thank you Julia for the feedback! :) I've updated testing, rebased the commits and made the commit msgs a bit better for this one.

davidbroughsmyth commented:
Your updates have been noted. LGTM.

rokane commented on 01_fib/marchitect/main_test.go:
```diff
@@ -0,0 +1,104 @@
+		if expected != actual {
+			t.Errorf("Unexpected output from test %s, expected = %s, actual = %s", testCase, expected, actual)
+		}
+
```
Little nit, but extra space here is inconsistent with the rest of the code

rokane reviewed (changes requested):
Looking really good. I only have one suggestion and that is to provide more descriptive names for your test scenarios. All of the scenarios are of the form "normal test x" but this doesn't describe what is being tested. Think about how these tests would look for someone else running them and try to make them as straight forward as possible.

An example below:

```go
// Output with the current version
--- FAIL: Test_abs (0.08s)
    --- PASS: Test_abs/normal_test_one (0.08s)
    --- PASS: Test_abs/normal_test_two (0.08s)
    --- FAIL: Test_abs/normal_test_three (0.08s)

# vs more descriptive names
--- FAIL: Test_abs (0.08s)
    --- PASS: Test_abs/Positive_Number (0.08s)
    --- PASS: Test_abs/Negative_Number (0.08s)
    --- FAIL: Test_abs/Zero_Value (0.08s)
```

If I was modifying this code, ran the tests and saw a failure (which is described above in the example) I would prefer to see a descriptive name as it would help me understand what I could have done wrong.

 Once this is done, please ping me and I will take another look asap.

anzdaddy commented:
Please close merge or close.
